its/src/test/java/com/sonar/it/scanner/msbuild/utils/ContextExtension.java

```diff
@@ -19,30 +19,45 @@
  */
 package com.sonar.it.scanner.msbuild.utils;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import org.junit.jupiter.api.extension.AfterEachCallback;
 import org.junit.jupiter.api.extension.BeforeEachCallback;
 import org.junit.jupiter.api.extension.ExtensionContext;
 
 public class ContextExtension implements BeforeEachCallback, AfterEachCallback {
 
   private static final ThreadLocal<String> currentTestName = new ThreadLocal<>();
+  private static final ThreadLocal<Path> currentTempDir = new ThreadLocal<>();
```

Comment:
The same as in https://github.com/SonarSource/sonar-scanner-msbuild/pull/2416#discussion_r2014499427 applies here.